File: prid-art-nor/backend/Models/Card.cs
using System;
using System.Linq;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Diagnostics;
using Microsoft.EntityFrameworkCore;
using System.Text.RegularExpressions;
using System.ComponentModel.DataAnnotations.Schema;

namespace prid_2021_A08.Models {

    public class Card {

        [Key]
        public int Id { get; set; }
        
        [Required, StringLength(20, MinimumLength = 3)]
        public string Name { get; set; }

        public int AuthorId { get; set; }

        public virtual List List { get; set; }

        public int ListId { get; set; }

        public virtual User Author { get; set; }

        public virtual IList<User> Participaters { get; set; } = new List<User>();

        public virtual IList<Participation> UserParticipations { get; set; } = new List<Participation>();

        public int Pos { get; set; }

        [NotMapped]
        public IEnumerable<int> ParticipatedUsers {
            get => UserParticipations.Select(p => p.UserId);
        }

    }

}

File: prid-art-nor/frontend/src/app/models/card.ts
import { User } from "./user";

export class Card {
    id?: number;
    name?: string;
    author?: User;
    authorId? : number;
    participaters?: number[];
    listId? : number;
    pos? : number;

    constructor(data: any) {
        if (data) {
          this.id = data.id;
          this.name = data.name;
          this.author = data.author;
          this.authorId = data.authorId;
          this.participaters = data.participaters;
          this.listId = data.listId;
          this.pos = data.pos;
        }
    }
}